optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -1406,6 +1524,8 @@ public void remove(Key key, Value value) {
      * <p>
      * Makes no guarantees on iteration order for map entries.
      * For stable iteration order, use {@link #toSortedMap(Function, Function, BinaryOperator)}.
+     * <p>
+     * This constraint collector is slow and causes considerable garbage collector pressure.
```

Comment:
If we do this change for toList and toSet and toSortedSet, we should do it for toMap etc too, for consistency (principle of least surprise).

And yes, I think we need to do this change :)

Reply:
Making `toMap()` incremental in this way is likely impossible.
The interface of the method requires post-processing on the internal map before the collector returns.

Reply:
https://issues.redhat.com/browse/PLANNER-2406